test/game/game_setup_test.exs

```diff
@@ -83,5 +83,18 @@ defmodule GameSetupTest do
       assert GameSetup.setup_players(@mock_command_line) ==
         %Players{player_one: %Player{type: HardComputerPlayer, symbol: "X"}, player_two: %Player{type: HardComputerPlayer, symbol: "O"}}
     end
+
+    test "displays invalid setup validation message via invalid input, then returns players containing selected players and associated symbols" do
+      user_inputs = ["2", "3", "2", "2", "2"]
+      start_supervised!({@mock_command_line.communicator, user_inputs})
+
+      players = GameSetup.setup_players(@mock_command_line)
+
+      assert :sys.get_state(CommunicatorMockServer) =~
```

Comment:
Disclaimer: This is unrelated to this PR, but I know you mentioned you were looking for refactor opportunities in your project.

Instead of relying on `:sys.get_state` to obtain what has been displayed to the console/interface let's consider implementing a public function on the `CommunicatorMock` that returns the state.

Reply:
Ill be sure to take note of this and create a PR for it after the tickets! I have a few refactors I want to do 😄  Thank you for the help! 🥳